### PR TITLE
feat: disable previous dates from now date

### LINKED ISSUE #840: Proposal creation: disable previous days on the calendar
When creating a proposal and setup start and end date of the proposal we should make the previous days not selectable. The day number should use the css var "text-color" example:
image

### PR DESCRIPTION
**Pull request by sarmatdev**
Fixes #840 .

Changes proposed in this pull request:
- Fix `isSelectable` algorithm in `UiCalendar` base component
- 
-

### CODE REVIEW

**Comment by bonustrack** [comment]
@sarmatdev please add the issue number in your PR description

File: src/components/Ui/Calendar.vue [hunk 1]
```diff
@@ -61,16 +61,11 @@ function toggleDay(year, month, day) {
...
-      return (
-        new Date(year, month, day) > new Date() &&
-        new Date(year, month, day) < in30Days
-      );
-      */
+function isSelectable(year, month, day) {
+  const dateForCheck = new Date(year, month, day);
+  const diff = new Date().setHours(0, 0, 0, 0) - dateForCheck;
+
+  return diff > 0 ? false : true;
```
Comment (samuveth) [inline src/components/Ui/Calendar.vue:68]: @sarmatdev wouldn't `return !(diff > 0)` do the same thing?

**Comment by sarmatdev** [comment]
> @sarmatdev can we also disable the back arrow when the previous month have already past.



> @sarmatdev can we also disable the back arrow when the previous month have already past.

Ready to check

**Comment by bonustrack** [comment]
@sarmatdev Can you remove the button back instead of disabling it?

**Comment by samuveth** [comment]
@sarmatdev the month and year are not longer centered 
image

**Comment by sarmatdev** [comment]
> @sarmatdev the month and year are not longer centered image

Oh, really... Any ideas how it might be centered?🤔
I have tried several, but I did not get accurate centering

File: src/components/Ui/Calendar.vue [hunk 2]
```diff
@@ -61,23 +64,19 @@ function toggleDay(year, month, day) {
...
-      const in30Days = new Date();
-      in30Days.setDate(in30Days.getDate() + 30);
-      return (
-        new Date(year, month, day) > new Date() &&
-        new Date(year, month, day) < in30Days
-      );
-      */
+function isSelectable(year, month, day) {
+  const dateForCheck = new Date(year, month, day);
+  const diff = new Date().setHours(0, 0, 0, 0) - dateForCheck;
```
Comment (midgerate) [inline src/components/Ui/Calendar.vue:69]: cant we use date-fns or moment to do these calculations? I find this strategy prone to mistakes. @bonustrack

Comment (samuveth) [reply src/components/Ui/Calendar.vue:69]: @midgerate can you provide an example on how that would be possible?

Comment (midgerate) [reply src/components/Ui/Calendar.vue:69]: @sarmatdev no mistake. We are contemplating to use Day.js its not a blocker though.

**Pull request closed without merging**